sympy/functions/special/bessel.py

```diff
@@ -207,19 +210,38 @@ def eval(cls, nu, z):
             newz, n = z.extract_branch_factor()
             if n != 0:
                 return exp(2*n*pi*nu*I)*besselj(nu, newz)
+
         nnu = unpolarify(nu)
         if nu != nnu:
             return besselj(nnu, z)
 
     def _eval_rewrite_as_besseli(self, nu, z, **kwargs):
-        return exp(I*pi*nu/2)*besseli(nu, polar_lift(-I)*z)
+        #return exp(I*pi*nu/2)*besseli(nu, polar_lift(-I)*z)
+        # 9.6.3 in Abramovitz and Stegun
+        argz = arg(z)
+        not_lower_left = ((-pi/2 < argz) & (argz <= pi))
+        return Piecewise(
+            (exp(+1*I*pi*nu/2)*besseli(nu, z*exp(-1*pi*I/2)), not_lower_left),
+            (exp(-3*I*pi*nu/2)*besseli(nu, z*exp(+3*pi*I/2)), True),
```

Comment:
This formulation fails for `z=0` because `arg(0)` gives `NaN`. I think maybe it's better to write the condition using `re` and `im` rather than `arg`:
```python
not_lower_left = (re(z) > 0) | (im(z) >= 0)
```